src/model/rnn.py

```diff
@@ -266,45 +268,89 @@ def __init__(
         self.output_recurrent_outputs = output_recurrent_outputs
         self.output_last_state = output_last_state
 
+
+# class RNNForLanguageModeling(nn.Module):
+class RNNForLanguageModeling(PreTrainedModel):
+    """RNN with optional dropout for use with a language modeling objective.
+
+    Consists of three components:
+        * Embedding layer
+        * Recurrent layer- choice of vanilla (Elman) RNN, LSTM, or GRU
+        * Language modeling head- linear layer with bias
+
+    Primarily intended for use in conjunction with a HuggingFace `Trainer` instance, or
+    at least with a `transformers.tokenization_utils.PreTrainedTokenizer` instance.
+    See the method documentation of `forward` for more information on how to use this
+    without a `PreTrainedTokenizer` or `Trainer`.
+
+    Class attributes:
+        config_class:
+            A subclass of [`PretrainedConfig`] to use as configuration class for this
+            model architecture.
+        base_model_prefix:
+            A string indicating the attribute associated to the base model in derived
+            classes of the same architecture adding modules on top of the base model.
+            Technically unnecesary at this juncture but may be necessary in future if
+            we make RNNForLanguageModeling a subclass of a new class RNN which
+            subclasses from PreTrainedModel (for example, if we also wanted to implement
+            a RNNForSequenceClassification or other RNN models)
+        _tied_weights_keys:
+            A list of `state_dict` keys that are potentially tied to another key in the
+            state_dict. I think only one set of the copies needs to be mentioned below?
+            hence omitting embedding.weight. If this breaks save_pretrained, may need
+            to revisit/revise.
+    """
+
+    config_class = RNNConfig
+    base_model_prefix = "model"
+    _tied_weights_keys = ["lm_head.weight"]
+
+    def __init__(self, config: RNNConfig):
+        super().__init__(config)
+
         self.embedding = nn.Embedding(
             num_embeddings=vocab_size, embedding_dim=embedding_dim, **embedding_kwargs
         )
 
         self.recurrent: nn.RNNBase = getattr(nn, RNNType(rnn_type.upper()))(
```

Comment:
Where is `rnn_type` defined? Does this need to be `config.rnn_type`? Not obvoius to me how this is working at all otherwise

Reply:
Good catch, and I thought I had already caught this; turns out I had not pushed that commit, however, which also includes a couple other minor but crucial changes (e.g. the actual config files). Updated now :)